sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -539,6 +539,25 @@ object PartitioningUtils {
     }).asNullable
   }
 
+  def requestedPartitionColumnIds(
```

Comment:
Can we have a more intuitive name? This function name looks weird to me because `requestedPartitionColumnIds` returns full schema.

Reply:
The returned value depends on the parameter `requiredSchema`. The parameter can be full schema or requested schema.
Do you have suggestion for the method name?

Reply:
If you don't mind, I prefer to revert the last commit, https://github.com/apache/spark/pull/23387/commits/1b58df82c6ac5580f14a3b68022b88915c577646 .